core/query/customlist.py

```diff
@@ -44,6 +44,21 @@ def share_locally_with_library(
         library_collection_ids = [c.id for c in library.collections]
         entry: CustomListEntry
         for entry in customlist.entries:
+            # It appears that many many lists have entries without works.
+            # see https://ebce-lyrasis.atlassian.net/browse/PP-708 for the full story.
+            # Because of this frequently occurring condition, lists are quietly not shared
+            # with the majority of libraries causing confusion for our users.  As it stands
+            # there is nothing that prevents lists with work-less entries that have already been
+            # shared from being unshared.  So for the time being the least intrusive intervention
+            # for enabling sharing to work again for many existing lists would be to relax the
+            # validation when an entry does not have an associated work.
+            if not entry.work:
+                log.warning(
```

Comment:
This might produce a lot of not really helpful log messages. Another approach would be to count this type of error and then emit a warning with the count, if the count is greater than zero.